manifests/init.pp

```diff
@@ -820,8 +820,77 @@
                 ]
               }
             }
+            '16.04': {
+              $default_pam_d_login_template = 'pam/login.ubuntu16.erb'
+              $default_pam_d_sshd_template  = 'pam/sshd.ubuntu16.erb'
+              $default_package_name         = 'libpam0g'
+
+              if $ensure_vas == 'present' {
+                $default_pam_auth_lines = [
```

Comment:
VAS is a commercial product. If you do not run it, then just doing the fail() is fine. Someone that uses it can add support if needed.

http://software.dell.com/products/authentication-services/